dwitter/templates/base.html

```diff
@@ -23,13 +23,46 @@
     {% load staticfiles %}
     <link href="https://fonts.googleapis.com/css?family=Roboto:400,400i,700,700i" rel="stylesheet">
     {% compress css %}
-    <link rel=stylesheet href='{% static "main.css" %}' >
+    <link id='style' rel=stylesheet href='{% static "main.css" %}' >
     {% endcompress %}
 
     {% block head %}
     {% endblock %}
   </head>
   <body class="{% block body_class %}{% endblock %}">
+    <script type="text/javascript">
+      // Check for compatibility
+      if (window.localStorage){
+        var l = document.getElementById('style');
+	var switchTheme = function(){
+	  // Change the imported stylesheet from main.css to main_dark.css or vice versa
+	  // Change localstorage theme to 'dark'
+	  var theme = localStorage.getItem('theme');
+	  if (theme == 'dark'){
+	    localStorage.setItem('theme','');
+	    l.setAttribute('href', '{% static "main.css" %}');
+	  }
+	  else{
+	    localStorage.setItem('theme','dark');
+	    l.setAttribute('href', '{% static "main_dark.css" %}');
+	  }
+	  location.reload();
+	  // Call this function in the li to change the mode
+	  }
+		
+	  // Do the same thing outside the function without changing localStorage to set correct theme
+	  var theme = localStorage.getItem('theme');
+	  if (theme == 'dark'){
+	    l.setAttribute('href', '{% static "main_dark.css" %}');
+	  }
+	  else{
+	    l.setAttribute('href', '{% static "main.css" %}');
+	  }
```

Comment:
I think this code looks nicer now. :)

If you like, you could make further reductions by 1. pulling repeated code (setAttribute) into a function (setTheme), and 2. using ternary (`a ? b : c`) instead of `if-then-else`.

(It's a matter of preference, but I usually use ternary for expressions, if-then-else for actions / imperative instructions.)

```js
    // Call this function from the li to change the mode
    var switchTheme = function () {
      var theme = localStorage.getItem('theme');
      localStorage.setItem(theme === 'dark' ? '' : 'dark');
      setTheme(); // BTW, do we really need this setTheme(), if we are about to reload the page?
      location.reload();
    };

    var setTheme = function () {
      var theme = localStorage.getItem('theme');
      l.setAttribute('href', theme === 'dark' ? '{% static "main_dark.css" %}' : '{% static "main.css" %}');
    };

    // When the page loads
    setTheme();
```